cpu/stm32f1/periph/adc.c

```diff
@@ -57,10 +57,28 @@ static inline void prep(adc_t line)
 {
     mutex_lock(&locks[adc_config[line].dev]);
     periph_clk_en(APB2, (RCC_APB2ENR_ADC1EN << adc_config[line].dev));
+
+    /* enable the ADC module */
+    dev(line)->CR2 |= ADC_CR2_ADON;
```

Comment:
Shouldn't this have been done during `adc_sample()` even before?

Reply:
I thought this would be the right place, because before the `prep()` function call in `adc_sample()`are only variable checks and no periphery related instructions. In `adc_init()` this is the same behaviour.

Reply:
I also think it is. I just wonder how this ever worked without.